feat(components/modals): remove 'string' from `SkyConfirmButton`'s `styleType` type

Blackbaud-ErikaMcVey opened the pull request:
BREAKING CHANGE: `SkyConfirmButton`'s `styleType` will only accept predefined strings of type `SkyConfirmButtonStyleType`. To address this, ensure `styleType` is only being set to a supported value.

Blackbaud-PatrickOFriel commented:
#627

Blackbaud-PaulCrowder commented:
Approved

[![Approved with PullApprove](https://img.shields.io/badge/pullapprove-approved-brightgreen.svg)](https://pullapprove.com/blackbaud/skyux/pull-request/664/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=blackbaud/skyux)

Blackbaud-ErikaMcVey committed:
feat(components/modals): remove 'string' from `SkyConfirmButton`'s `styleType` type

BREAKING CHANGE: `SkyConfirmButton`'s `styleType` will only accept predefined strings of type
`SkyConfirmButtonStyleType`. To address this, ensure `styleType` is only being set to a supported
value.

blackbaud-sky-build-user commented:
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/664/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/664/storybooks/ag-grid-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/664/storybooks/layout-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/664/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/664/storybooks/pages-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/664/storybooks/tabs-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/664/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/664/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/664/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/664/playground/)

Blackbaud-ErikaMcVey commented:
[AB#2315249](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2315249)
